plugins/org.jboss.tools.livereload.core/src/org/jboss/tools/livereload/core/internal/util/PathBuilder.java

```diff
@@ -0,0 +1,44 @@
+package org.jboss.tools.livereload.core.internal.util;
+
+/**
+ * Convenient utility class to build URL path and make sure {@code /} separators
+ * are properly handled.
+ */
+public class PathBuilder {
+
+	private final StringBuilder path = new StringBuilder();
+
+	public static PathBuilder from(final String pathFragment) {
+		final PathBuilder pathBuilder = new PathBuilder();
+		return pathBuilder.path(pathFragment);
+	}
+
+	public PathBuilder path(final String pathFragment) {
+		if (pathFragment == null) {
+			return this;
+		}
+		if (!pathFragment.startsWith("/") && !currentPathEndsWith('/')) { 
```

Comment:
in terms of code readability it would have been easier if the cases were grouped (conditions keep being repeated): 
if (pathFragment.startsWith("/") {
  if (currentPathEndsWith('/') {
   ...
  } else {
  }
} else {
 ...
}